test/simulator/misc/random-teams.js

```diff
@@ -26,6 +27,12 @@ function isValidSet(gen, set) {
 	return true;
 }
 
+function validLearnset(move, set, tier) {
+	const validator = TeamValidator(`gen7${tier}`);
+	const template = validator.dex.getTemplate(set.species || set.name);
+	return !validator.checkLearnset(move, template);
+}
+
 describe(`Random Team generator`, function () {
 	for (const gen of ALL_GENS) {
 		it(`should successfully create valid Gen ${gen} teams`, function () {
```

Comment:
This makes for ludicrously long tests. Add ``this.timeout(0)`` below.

Reply:
I mean.... we don't really want to have a test suite with ludicrously long tests either though....

Reply:
Then the test can be conditionally run depending on the arguments passed to ``mocha``.

Personally, I like to unlock long tests by passing ``-t NUM``, which also sets the timeout for tests. We could have that as ``npm run fulltest``

Reply:
having additional options for long tests seems like it would be used (selfishly thinking of my smoke test use case)

Reply:
In doing this, other tests cause to fail, such as Intimidate in gen 4, and endless battle clause.